pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -24,18 +24,33 @@ import (
 	"fmt"
 
 	"k8s.io/api/admission/v1beta1"
+	authenticationv1 "k8s.io/api/authentication/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
 
 	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	cdiwebhook "kubevirt.io/containerized-data-importer/pkg/apiserver/webhooks/api"
 	"kubevirt.io/kubevirt/pkg/virt-api/webhooks"
 	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
 )
 
 var validRunStrategies = []v1.VirtualMachineRunStrategy{v1.RunStrategyHalted, v1.RunStrategyManual, v1.RunStrategyAlways, v1.RunStrategyRerunOnFailure}
 
+type CloneAuthFunc func(namespace, name string, userInfo authenticationv1.UserInfo) (bool, string, error)
+
 type VMsAdmitter struct {
 	ClusterConfig *virtconfig.ClusterConfig
+	cloneAuthFunc CloneAuthFunc
+}
+
+func NewVMsAdmitter(clusterConfig *virtconfig.ClusterConfig, client kubecli.KubevirtClient) *VMsAdmitter {
+	return &VMsAdmitter{
+		ClusterConfig: clusterConfig,
+		cloneAuthFunc: func(namespace, name string, userInfo authenticationv1.UserInfo) (bool, string, error) {
+			return cdiwebhook.CanClonePVC(client, namespace, name, userInfo)
```

Comment:
This is kind of clever, I like that we're not exposing the kubecli.KubevirtClient to be used just anywhere in the validation code, instead it's completely isolated to the cloneAuthFunc :+1: pretty neat idea.